src/commands/inhouse.js

```diff
@@ -0,0 +1,271 @@
+											.addField(joinedNames.join(', '), false)
+									).then(respone => {
+										serverInhouseMessageIDs[message.guild.name] = respone.id;
+									});
+								} else {
+									message.delete();
+									message.channel.fetchMessage(serverInhouseMessageIDs[message.guild.name])
+										.then(message => message.edit('',
+											new Discord.RichEmbed({title: 'Current Players:'})
+												.addField(joinedNames.join(', '), false)));
+								}
+							}
+						});
+					} else {
+						message.channel.send('Please include your League of Legends summoner name like so: !f inhouse join <summoner name>');
+					}
+					break;
+				} 
+			} catch (err) {
+				logger.error(err);
+			}
+		}
+	} };
+
+class Inhouse {
+	constructor(_type) {
+		this.players = [];
+		this.currentMatch = null;
+		this.type = _type;
+	}
+
+	addPlayerToInhouse(newPLayerDiscordName, newPlayerSummonerName) {
+		return new Promise((resolve) =>{
+			new Player(newPLayerDiscordName, newPlayerSummonerName).then(response => {
+				let newPlayer = response;
+				if (newPlayer.summonerID === -1) {
+					resolve(-1);
+				}             
+				this.players.push(newPlayer);
+				if (this.players.length === 10 && this.currentMatch === null) {
+					this.CreateMatch();
+					resolve(0);
+				}
+    
+				resolve(1);
+			});
+		});   
+	}
+
+	CreateMatch() {
+		this.currentMatch = new Match(this.players, this.type);
+	}
+}
+
+class Player {
+	constructor(playerDiscordName, playerSummonerName) {
+		return new Promise((resolve) => {
+			this.discordName = playerDiscordName;
+			this.summonerName = playerSummonerName;
+			this.summonerID;
+			this.rankWeight;
+    
+			this.getSummonerId(this.summonerName).then(response => {
+				this.summonerID = response;
+				if (this.summonerID !== null) {
+					this.getRankWeight(this.summonerID).then(response => {
+						this.rankWeight = response;
+						resolve(this);
+					});
+				}
+			});
+		});
+
+	}
+
+	getSummonerId(name) {
+		return new Promise((resolve, reject) => {
+			let _summonerId;
+			let fullUrl = urlSummId + name + urlFinish;
+        
+			https.get(fullUrl, (resp) => {
+				let data = '';
+				resp.on('data', (chunk) => {
+					data += chunk;
+				});
+				resp.on('end', () => {
+					try {
+						let jsonResp = JSON.parse(data);
+						_summonerId = jsonResp.id;
+						resolve(_summonerId);
+					} catch (e) {
+						reject(e.message);
+					}
+				});
+			}).on('error', (err) => {
+				reject(`Got error: ${err.message}`);
+			});
+		});
+	}
+    
+	getRankWeight(summId) {
+		return new Promise((resolve, reject) => {
+			let _rankWeight = 0;
+			let tier;
+			let division;
+    
+			let fullUrl = urlSummRank + summId + urlFinish;
+			https.get(fullUrl, (resp) => {
+				let data = '';
+				resp.on('data', (chunk) => {
+					data += chunk;
+				});
+				resp.on('end', () => {
+					try {
+						let jsonResp = JSON.parse(data);
+						for (let i = 0; i < 3; i++) {
```

Comment:
Just realized we can simplify this code. 
```javascript
try {
  let jsonResp = JSON.parse(data);
  jsonResp.forEach((curr) => {
    if (curr != null && curr.queueType === 'RANKED_SOLO_5x5') {
      tier = curr.tier;
      division = curr.rank;
      break
    }
  })
  _rankWeight = utils.calculateSummonerRankWeight(tier, division);
```

Reply:
Kept as for loop but simplified if check/added loop safety for jsonResp.length